upgrade/scripts/common/ncn-common.sh

```diff
@@ -87,128 +85,145 @@ export STABLE_NCN=$(hostname)
-    [ $? -ne 0 ] && return 1
-    ssh-keygen -R "${ncn_ip}" -f "${known_hosts}" > /dev/null 2>&1
-    [ $? -ne 0 ] && return 1
-    ssh-keyscan -H "${target_ncn},${ncn_ip}" > /dev/null 2>&1 >> "${known_hosts}"
-    res=$?
-
-    # remove the old authorized_hosts entry for the target NCN cluster-wide
-    {
-        NCNS=$(grep -oP 'ncn-w\w\d+|ncn-s\w\d+' /etc/hosts | sort -u)
-        HOSTS=$(echo $NCNS | tr -t ' ' ',')
-        pdsh -w $HOSTS ssh-keygen -R ${target_ncn}
-        pdsh -w $HOSTS ssh-keygen -R ${ncn_ip}
-    } >& /dev/null
-
-    set -e
-    return $res
+  set +e
+  local target_ncn ncn_ip known_hosts
+  known_hosts="/root/.ssh/known_hosts"
+  sed -i 's@pdsh.*@@' $known_hosts
+  target_ncn="$1"
+  ncn_ip=$(host ${target_ncn} | awk '{ print $NF }')
+  [[ -n ${ncn_ip} ]]
+  # Because we run with set +e in this function, check return codes after running commands
+  [ $? -ne 0 ] && return 1
+  echo "Updating SSH keys for node ${target_ncn} with IP address of ${ncn_ip}"
+  ssh-keygen -R "${target_ncn}" -f "${known_hosts}" > /dev/null 2>&1
+  [ $? -ne 0 ] && return 1
+  ssh-keygen -R "${ncn_ip}" -f "${known_hosts}" > /dev/null 2>&1
+  [ $? -ne 0 ] && return 1
+  ssh-keyscan -H "${target_ncn},${ncn_ip}" > /dev/null 2>&1 >> "${known_hosts}"
+  res=$?
+
+  # remove the old authorized_hosts entry for the target NCN cluster-wide
+  {
+    NCNS=$(grep -oP 'ncn-w\w\d+|ncn-s\w\d+' /etc/hosts | sort -u)
+    HOSTS=$(echo $NCNS | tr -t ' ' ',')
+    pdsh -w $HOSTS ssh-keygen -R ${target_ncn}
+    pdsh -w $HOSTS ssh-keygen -R ${ncn_ip}
+  } >&/dev/null
+
+  set -e
+  return $res
 }
 
 function wait_for_kubernetes() {
   local target_ncn=$1
   state_name="WAIT_FOR_K8S"
   state_recorded=$(is_state_recorded "${state_name}" ${target_ncn})
   if [[ $state_recorded == "0" ]]; then
-      echo "====> ${state_name} ..."
-      {
+    echo "====> ${state_name} ..."
+    {
       set +e
       echo "waiting for k8s: $target_ncn ..."
-      until csi automate ncn kubernetes --action is-member --ncn $target_ncn --kubeconfig /etc/kubernetes/admin.conf
-      do
-          sleep 5
+      until csi automate ncn kubernetes --action is-member --ncn $target_ncn --kubeconfig /etc/kubernetes/admin.conf; do
+        sleep 5
       done
       # Restore set -e
       set -e
       echo "$target_ncn joined k8s"
-      } >> ${LOG_FILE} 2>&1
+    } >> ${LOG_FILE} 2>&1
+
+    record_state "${state_name}" ${target_ncn}
+  else
+    echo "====> ${state_name} has been completed"
+  fi
+}
 
-      record_state "${state_name}" ${target_ncn}
+function update_test_rpms() {
+  local state_name state_recorded target_ncn
+  target_ncn=$1
+  state_name="UPDATE_TEST_RPMS"
+  state_recorded=$(is_state_recorded "${state_name}" ${target_ncn})
+  if [[ $state_recorded == "0" ]]; then
+    echo "====> ${state_name} ..."
+    {
+      ssh $target_ncn '/usr/share/doc/csm/upgrade/scripts/upgrade/util/upgrade-test-rpms.sh --local'
```

Comment:
If this is ran after the node rebuilds, then we don't need it here. Cloud-init will installs these, but we have to add craycli and hpe-csm-goss-package to https://github.com/Cray-HPE/csm/blob/release/1.6/rpm/cloud-init.yaml#L28